### PR TITLE
add DingTalk chatroom robot notification support

### PR DESCRIPTION
**Pull request by wuutiing**
[DingTalk](https://www.dingtalk.com/en) is a common used chat app in China, more than 10 million companies and 3 quarters of billion people are using or have used DingTalk. here i've done an implementation for DingTalk support, hope you accept it.
a test room is available, search 30005872 in DingTalk.
img

### CODE REVIEW

File: knockknock/dingtalk_sender.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    DingTalk sender wrapper: execute func, send a DingTalk notification with the end status
+    (sucessfully finished or crashed) at the end. Also send a DingTalk notification before
+    executing func.
+
+    `webhook_url`: str
+        The webhook URL to access your DingTalk chatroom.
+        Visit https://ding-doc.dingtalk.com/doc#/serverapi2/qf2nxq for more details.
+    `user_mentions`: List[str] (default=[])
+        Optional users phone number to notify.
+        Visit https://api.slack.com/methods/users.identity for more details.
```
Comment (VictorSanh) [inline knockknock/dingtalk_sender.py]: mind refining that comment?

File: README.md [hunk 2]
```diff
@@ -260,6 +261,36 @@ knockknock matrix \
...
+### DingTalk
+DingTalk is now supported thanks to [@wuutiing](https://github.com/wuutiing). Given DingTalk chatroom robot's webhook url and secret/keywords(at least one of them are set when creating a chatroom robot), your notifications will be sent to reach any one in that chatroom.
+
+#### Python
+
+```python
+from knockknock import dingtalk_sender
+
+webhook_url = "<webhook_url_to_your_dingtalk_chatroom_robot>"
+@slack_sender(webhook_url=webhook_url, secret="<your_robot_secret_if_set>", keywords=["<list_of_keywords_if_set>"])
```
Comment (VictorSanh) [inline README.md]: Small typo: `dingtalk_sender` instead of `slack_sender `

**Comment by VictorSanh** [comment]
I had a look, it looks good to me, I'm merging it as soon as you precized the doc in `dingtalk_sender.py` (I'm not sure how to modify it so I prefer that you do it).
Thanks @wuutiing ! Sorry it took so long.

File: README.md (hunk 2 above)
Comment (wuutiing) [reply README.md]: > Small typo: `dingtalk_sender` instead of `slack_sender `

sure, fixed

File: knockknock/dingtalk_sender.py (hunk 1 above)
Comment (wuutiing) [reply knockknock/dingtalk_sender.py]: done

**Comment by wuutiing** [comment]
> I had a look, it looks good to me, I'm merging it as soon as you precized the doc in `dingtalk_sender.py` (I'm not sure how to modify it so I prefer that you do it).
> Thanks @wuutiing ! Sorry it took so long.

Sorry reply late. And have fixed the inference url and readme typo, looking for your comments.

**Pull request merged**